src/v/storage/disk_log_impl.cc

```diff
@@ -557,46 +558,60 @@ model::term_id disk_log_impl::term() const {
     }
     return _segs.back()->offsets().term;
 }
+static offset_stats empty_offsets(model::offset start_offset) {
+    offset_stats ret;
+    ret.start_offset = start_offset;
+    if (ret.start_offset > model::offset(0)) {
+        ret.dirty_offset = ret.start_offset - model::offset(1);
+        ret.committed_offset = ret.dirty_offset;
+    }
+    return ret;
+}
 
 offset_stats disk_log_impl::offsets() const {
     if (_segs.empty()) {
-        offset_stats ret;
-        ret.start_offset = _start_offset;
-        if (ret.start_offset > model::offset(0)) {
-            ret.dirty_offset = ret.start_offset - model::offset(1);
-            ret.committed_offset = ret.dirty_offset;
-        }
-        return ret;
-    }
-    // NOTE: we have to do this because ss::circular_buffer<> does not provide
-    // with reverse iterators, so we manually find the iterator
-    segment_set::type end;
-    segment_set::type term_start;
-    for (int i = (int)_segs.size() - 1; i >= 0; --i) {
-        auto& seg = _segs[i];
-        if (!seg->empty()) {
-            if (!end) {
-                end = seg;
-            }
-            // find term start offset
-            if (seg->offsets().term < end->offsets().term) {
-                break;
-            }
-            term_start = seg;
-        }
+        return empty_offsets(_start_offset);
```

Comment:
nit: This early check isn't needed, it's covered after the non-empty reverse iterators.